purchases/src/main/java/com/revenuecat/purchases/Purchases.kt

```diff
@@ -390,38 +390,72 @@ class Purchases @JvmOverloads internal constructor(
         }
     }
 
-    override fun onPurchasesUpdated(purchases: List<Purchase>) {
+    private fun afterSetListener(value: PurchasesListener?) {
+        if (value != null) {
+            billingWrapper.setListener(this)
+            application.registerActivityLifecycleCallbacks(this)
+            getCaches()
+            restorePurchasesForPlayStoreAccount()
```

Comment:
I think we might need to rething this. It's causing confusion.

Reply:
@jeiting do you mean for this PR? what exactly is causing confusion?